src/dmn/trn_transit_xdp_usr.c

```diff
@@ -639,39 +639,58 @@ uint32_t trn_get_interface_ipv4(int itf_idx)
 
 int trn_update_transit_network_policy_primary_map(struct user_metadata_t *md,
 						  struct vsip_cidr_t *ipcidr,
-						  __u64 bitmap)
+						  __u64 *bitmap)
 {
-	int err = bpf_map_update_elem(md->ing_vsip_prim_map_fd, ipcidr, &bitmap, 0);
-	if (err) {
-		TRN_LOG_ERROR("Store Primary ingress map failed (err:%d).",
-			      err);
-		return 1;
+	int counter = (int)(sizeof(ipcidr) / sizeof(*ipcidr));
```

Comment:
whill this get desired value of element number? I am afraid not.

Reply:
I tried using int pointer, it did return the correct value. But I can change to sizeof(struct vsip_cidr_t)
https://stackoverflow.com/questions/1898657/result-of-sizeof-on-array-of-structs-in-c

Reply:
the problem is like this: 
struct foo{...};
struct foo *pFoo;
sizeof(pFoo) is always the length of pointer (8 in ,u machine).

Reply:
sizeof(pFoo) is the length of pointer. sizeof(*pFoo) is the length of the first element. sizeof(pFoo)/sizeof(*pFoo) is the number of elements. 

    int chars[20];
    int count = sizeof(chars);
    printf("%d \n %ld", count, sizeof(int));

this outputs 80 4